packages/sitecore-jss-react/src/components/BYOCComponent.tsx

```diff
@@ -100,36 +96,25 @@ export class BYOCRenderer extends React.Component<BYOCRendererProps> {
       );
     }
     const { ComponentName: componentName } = props.params || {};
-    if (!componentName) {
-      const noNameProps = {
-        errorOverride: 'BYOC: The ComponentName for this rendering is missing',
-      };
-      return props.missingComponentComponent ? (
-        <this.props.missingComponentComponent {...noNameProps} />
-      ) : (
-        <MissingComponent {...noNameProps} />
-      );
-    }
-    // props.components would contain component from internal FEAAS regsitered component collection (registered in app)
-    // we can't access this collection here directly, as the collection from packages's dependency would be different from the one in app
-    const Component = props.components[componentName]?.component;
 
-    if (!Component) {
-      console.warn(
-        `Component "${componentName}" was not registered, please ensure the FEEAS.External.registerComponent call is made.`
-      );
-      const missingProps = {
-        rendering: {
-          componentName: componentName,
-        },
-        errorOverride: 'BYOC: This component was not registered.',
-      };
-      return props.missingComponentComponent ? (
-        <this.props.missingComponentComponent {...missingProps} />
-      ) : (
-        <MissingComponent {...missingProps} />
-      );
-    }
+    const noNameProps = {
+      errorOverride: 'BYOC: The ComponentName for this rendering is missing',
+    };
+
+    const unRegisteredComponentProps = {
+      rendering: {
+        componentName: componentName,
+      },
+      errorOverride: 'BYOC: This component was not registered.',
+    };
+
+    const missingProps = !componentName ? noNameProps : unRegisteredComponentProps;
+
+    const missingComponent = props.missingComponentComponent ? (
```

Comment:
Still doesn't look like we return immediately if `componentName` is missing. It shouldn't attempt to parse ComponentProps or render <FEAAS.ExternalComponent/> in this case.

Reply:
Actually I asked a question following your last comment. Nvm, it's clear now.